### PR TITLE
feat(TMDB): add presence

### PR DESCRIPTION
**Pull request by keksiqc**
**Describe the changes in this pull request:**
Added `TMDB` presence

<details>
<summary> Proof showing the creation/modification is working as expected </summary>

<br />

<br />

<br />

<br />
and more
</details>

### CODE REVIEW

File: websites/T/TMDB/presence.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+				.textContent.trim()}`;
+		} else presenceData.details = "Viewing Popular People list";
+	} else if (pathname.startsWith("/discuss"))
+		presenceData.details = "Viewing discussions";
+	else if (pathname.startsWith("/leaderboard"))
+		presenceData.details = "Viewing leaderboard";
+	else if (pathname.startsWith("/talk"))
+		presenceData.details = "Viewing TMDB support";
+	else if (pathname.startsWith("/search")) {
+		const query = (document.querySelector("input") as HTMLInputElement).value;
```
Comment (EncryptedDev) [inline websites/T/TMDB/presence.ts:98]: Use Type Genetic 
```suggestion
		const query = (document.querySelector<HTMLInputElement>("input")).value;
```

File: websites/T/TMDB/presence.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
+const presence: Presence = new Presence({
+		clientId: "966643093308837938"
+	}),
+	startTimestamp: number = Math.floor(Date.now() / 1000),
```
Comment (EncryptedDev) [inline websites/T/TMDB/presence.ts:4]:
```suggestion
	browsingTimestamp = Math.floor(Date.now() / 1000),
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/T/TMDB/presence.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
+const presence: Presence = new Presence({
+		clientId: "966643093308837938"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	strings = presence.getStrings({
+		browsing: "presence.activity.browsing",
+		reading: "presence.activity.reading"
+	});
```
Comment (Slowlife01) [inline websites/T/TMDB/presence.ts:8]: You seems to only be using 1 string? Just remove this and hardcode it.

File: websites/T/TMDB/presence.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+			presenceData.details = "Viewing top rated movies";
+		else if (pathname.startsWith("/movie/new"))
+			presenceData.details = "Adding a new movie";
+		else if (pathname.split("/").length === 3) {
+			presenceData.details = `Viewing ${document
+				.querySelector(`h2 > [href="${pathname}"]`)
+				.textContent.trim()}`;
+		} else if (pathname.includes("/reviews")) {
+			const pathArr = pathname.split("/");
+			pathArr.length = 3;
```
Comment (Slowlife01) [inline websites/T/TMDB/presence.ts:52]:
```suggestion
			const pathArr = pathname.split("/").slice(0,3);
```

File: websites/T/TMDB/presence.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+			presenceData.details = "Viewing top rated TV shows";
+		else if (pathname.startsWith("/tv/new"))
+			presenceData.details = "Adding a new TV show";
+		else if (pathname.split("/").length === 3) {
+			presenceData.details = `Viewing ${document
+				.querySelector(`h2 > [href="${pathname}"]`)
+				.textContent.trim()}`;
+		} else if (pathname.includes("/reviews")) {
+			const pathArr = pathname.split("/");
+			pathArr.length = 3;
```
Comment (Slowlife01) [inline websites/T/TMDB/presence.ts:76]:
```suggestion
			const pathArr = pathname.split("/").slice(0,3);
```

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

File: websites/T/TMDB/presence.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+	else if (pathname.startsWith("/search")) {
+		const query = document.querySelector<HTMLInputElement>("input").value;
+		if (query.length > 0) presenceData.details = `Searching for ${query}`;
+		else presenceData.details = "Searching";
+	} else if (pathname.startsWith("/")) {
+		presenceData.details = (
+			await presence.getStrings({
+				browsing: "presence.activity.browsing"
+			})
+		).browsing;
```
Comment (Slowlife01) [inline websites/T/TMDB/presence.ts:106]: This is what I meant
```ts
presenceData.details = "Browsing";
```

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

File: websites/T/TMDB/presence.ts (hunk 6 above)
Comment (keksiqc) [reply websites/T/TMDB/presence.ts:106]: oh yes sounds more logical, I am a bit tired sorry

**Comment by EncryptedDev** [comment]
Thought rebasing would fix the format issue, apparently not. I'll fix it soon :)

**Comment by Slowlife01** [comment]
whoops, can you open a new pull request?

**Pull request closed without merging**